src/agent.rs

```diff
@@ -92,22 +193,41 @@ impl CodexAgent {
         &self,
         session_id: &acp::SessionId,
     ) -> Result<Arc<CodexConversation>, Error> {
-        if !self
-            .sessions
-            .borrow()
-            .contains_key(&session_id.0.to_string())
-        {
-            return Err(Error::invalid_params().with_data("session not found"));
+        let maybe_state = self.sessions.borrow().get(session_id.0.as_ref()).cloned();
```

Comment:
If we have already saved conversation_id and conversation in `SessionState`, we can directly obtain them from `SessionState` here. However, the current implementation is a bit redundant.